api/fs/watcher.js

```diff
@@ -0,0 +1,241 @@
+import { EventEmitter } from '../events.js'
+import { AbortError } from '../errors.js'
+import { rand64 } from '../crypto.js'
+import { Buffer } from '../buffer.js'
+import hooks from '../hooks.js'
+import ipc from '../ipc.js'
+import gc from '../gc.js'
+
+/**
+ * Encodes filename based on encoding preference.
+ * @ignore
+ * @param {Watcher} watcher
+ * @param {string} filename
+ * @return {string|Buffer}
+ */
+function encodeFilename (watcher, filename) {
+  if (!watcher.encoding || watcher.encoding === 'utf8') {
+    return filename.toString()
+  }
+
+  if (watcher.encoding === 'buffer') {
+    return Buffer.from(filename.toString())
+  }
+
+  return filename
+}
+
+/**
+ * Starts the `fs.Watcher`
+ * @ignore
+ * @param {Watcher} watcher
+ * @return {Promise}
+ */
+async function start (watcher) {
+  const result = await ipc.send('fs.watch', {
+    path: watcher.path,
```

Comment:
should we double check these parameters exist? defaulting to something here seems like a possible foot-gun.

Reply:
they come from the `Watcher` instance - I can verify there because this function is private